lib/logstash/inputs/gelf.rb

```diff
@@ -90,12 +112,83 @@ def stop
   rescue IOError # the plugin is currently shutting down, so its safe to ignore theses errors
   end
 
+  private
+  def tcp_listener(output_queue)
+
+    @logger.warn("Starting gelf listener (tcp) ...", :address => "#{@host}:#{@port_tcp}")
+
+    if @tcp.nil?
+      @tcp = TCPServer.new(@host, @port_tcp)
+    end
+
+    while !@shutdown_requested
+      Thread.new(@tcp.accept) do |client|
+        @logger.debug? && @logger.debug("Gelf (tcp): Accepting connection from:  #{client.peeraddr[2]}:#{client.peeraddr[1]}")
+
+        begin
+          while !client.nil? && !client.eof?
+          
+            begin # Read from socket
+              @data_in = client.gets("\u0000")
```

Comment:
there's no need to make `data_in` an instance variable, we can remove the `@` here

Reply:
Done